lib/spack/spack/url_buildcache.py

```diff
@@ -1335,25 +1336,35 @@ class MirrorURLAndVersion:
     url: str
     version: int
 
-    def __init__(self, url: str, version: int):
+    def __init__(self, url: str, version: int, view: Optional[str] = None):
         self.url = url
         self.version = version
+        self.view = view
 
     def __str__(self):
-        return f"{self.url}__v{self.version}"
+        s = f"{self.url}__v{self.version}"
+        if self.view:
+            s += f"__@{self.view}"
+        return s
 
     def __eq__(self, other):
         if isinstance(other, MirrorURLAndVersion):
-            return self.url == other.url and self.version == other.version
+            return all(
+                [self.url == other.url, self.version == other.version, self.view == other.view]
+            )
```

Comment:
Please don't allocate an array to compare three properties.


```suggestion
            return self.url == other.url and self.version == other.version and self.view == other.view
```